cvmfs/cvmfs_server

```diff
@@ -4046,8 +4052,20 @@ snapshot() {
       -i ${spool_dir}/tmp/last_snapshot \
       -o .cvmfs_last_snapshot"
 
-    # remove magic file after successful replication
-    $user_shell "$delete_command"
+    # run the automatic garbage collection (if configured)
+    if is_garbage_collectable $name && [ ! -z "$CVMFS_GC_TIMESTAMP_THRESHOLD" ]
```

Comment:
Coming back to the configuration variables... There is already a sensitive default of '3 days ago' when you create a stratum1 from a garbage collectible repository. Hence, `cvmfs_server add-replica <...> <...>` will place `CVMFS_AUTO_GC_TIMESPAN='3 days ago'` in the config automatically.

We see the garbage collection as an experimental feature addressing mainly the nightly integration build use-case for now. The implementation and configuration of this features will almost certainly be subject to change in future versions. I suggest that we should first gain some experience with it and later see which changes are really necessary.